File: src/Core/UWP/Sakuno.ING.Views.UWP.Common/Controls/LoSIndicator.vb
Imports Sakuno.ING.Game.Models
Imports Sakuno.ING.ViewModels.Homeport

Namespace Controls
    Public NotInheritable Class LoSIndicator
        Inherits Control

        Public Sub New()
            DefaultStyleKey = GetType(LoSIndicator)
        End Sub

        Public ReadOnly Property ViewModel As New FleetLoSVM

        Public Property Effective As LineOfSight
            Get
                Return ViewModel.Effective
            End Get
            Set(value As LineOfSight)
                ViewModel.Effective = value
            End Set
        End Property

        Public Property Simple As Integer
            Get
                Return ViewModel.Simple
            End Get
            Set(value As Integer)
                ViewModel.Simple = value
            End Set
        End Property
    End Class
End Namespace

File: src/Core/UWP/Sakuno.ING.Views.UWP.Common/Controls/LoSIndicator.cs
using Sakuno.ING.Game.Models;
using Sakuno.ING.ViewModels.Homeport;
using Windows.UI.Xaml.Controls;

namespace Sakuno.ING.Views.UWP.Controls
{
    public sealed class LoSIndicator : Control
    {
        public LoSIndicator()
        {
            DefaultStyleKey = typeof(LoSIndicator);
        }

        public FleetLoSVM ViewModel { get; } = new FleetLoSVM();

        public LineOfSight Effective
        {
            get => ViewModel.Effective;
            set => ViewModel.Effective = value;
        }

        public int Simple
        {
            get => ViewModel.Simple;
            set => ViewModel.Simple = value;
        }
    }
}
